addons/account/models/account_move.py

```diff
@@ -1892,6 +1899,18 @@ def _compute_taxes_legal_notes(self):
                 if not is_html_empty(tax.invoice_legal_notes)
             )
 
+    @api.depends('line_ids.payment_date', 'line_ids.reconciled')
+    def _compute_next_payment_date(self):
+        for move in self:
+            move.next_payment_date = min([line.payment_date for line in move.line_ids.filtered(lambda l: l.payment_date and not l.reconciled)], default=False)
+
+    def _search_next_payment_date(self, operator, value):
+        lines_before_date = self.env['account.move.line'].search([
```

Comment:
```suggestion
        lines_before_date = self.env['account.move.line']._search([
```

Passing a sub query instead of a potentially huge list of ids should improve the performances by quite a lot

But maybe we don't expect the list to be huge and this sub query is always fast, and then this version is better?